### DISCUSSION TITLE
ttyUSB0

### DISCUSSION THREAD

**Discussion by BlackAlgae** [aqualinkd/AqualinkD#354] [Q&A]
Hey sfeakes/gang, 

I finally installed my donated rs8, I'm stuck at the "add /dev/ to it for complete path /dev/ttyUSB0" Here's some pictures of my pi zero, I apologize for not taking screenshots instead. 

The usb2rs485 kept turning the raspberry off till I swapped the powerbox to the charger. Then it wouldn't read until I wired the ground cable to the rs485. This is where I find myself now, thank you in advance!
![IMG_8258](https://github.com/user-attachments/assets/3e9bfe04-b534-4b95-aa19-0445f171f006)
![IMG_8259](https://github.com/user-attachments/assets/bc91b29d-fcd4-4a20-8eab-58d69e192688)

**Comment by sfeakes** [comment]
In the first screen shot you have not configured aqualinkd.  Information is in the wiki.
Second the usb2rs485 connector keeps disconnecting.  Not sure why as you have not provided details.  But until you fix that aqualinkd will fail for different reasons.
Have you gone over the wiki on how to connect to the usb2rs485 dongle to the panel?

**Reply by BlackAlgae** [reply]
Is configuring done through editing aqualinkd.config file? I read the whole wiki im just new to this and using terminal in general. With the usb2rs485 I had the issue where it wouldn't show me the "ttyusb0" had connected with only the data cable (yellow, black) but once i connected the green cables it started giving me this which youre saying keeps disconnecting. Maybe i flip the date cables.

How do i path it once it stops disconnecting and i have configured it. Also with configuring i have to set which jandy board im using (rs-8) right? Ill go over the wiki again but is there something else i should be configuring.

oh by usb2rs485 dongle what dop you mean? My usb2rs485 has 6 colored cables, the GYBR + brown and orange. Is the dongle the red 4 port rs485 inside the board? Thanks!

**Comment by sfeakes** [comment]
Yes aqualinkd is configured by editing /etc/aqualinkd.conf.
The device file /dev/ttyUSB0 will be created and deleted every time the USB2RS485 connector is connected or dis-connected. So if the file isn’t there, it’s been disconnected.  Since your logs show is connecting/disconnecting then that file will be created / deleted accordingly.
You should only need to connect 2 wired (data + and data -) to the USB2RS485 connector.  And if you have these wires connected or not, should not make a difference to if the pi connects / disconnects the USB itself.  Ie just plugging in the USB2RS485 connector to the Pi you should get /dev/ttyUSB0, it should not matter if you have the wires connected.

Power supply to the Pi is important, if the Pi sees under voltage then it will not power things like the USB2RS485 adapter, so you might have a power supply problem.

By dongle, yes I did mean the USB2RS485 connector.  As for wire colors, they can change on different installations, but the wiki will tell you the appropriate terminals, and again you only need to connect 2 wires.  This section https://github.com/sfeakes/AqualinkD/wiki#RS485

In the wiki, it will tell you to run some other commands to test the adapter, section following the one linked above.

Once you have got that far, you can run the serial_logger to give you the main config options to AqualinkD to connect to your panel.

**Reply by BlackAlgae** [reply]
Hey Sfeaks, thank you so much ill give another go! ill make sure im getting good power to the pi to get /dev/ttyUSB0. ill run the commands to test the adapte. Im still a bit confused as to what needs to be done on the aqualinkd.conf. but ill go over the wiki. And last question how do I run the serial_logger? ill make sure to go over the wiki 100 times to prevent these silly questions! thank you so much for replying to me im really excited about this project.
